source/core/src/main/com/csse3200/game/screens/SnakeScreen.java

```diff
@@ -0,0 +1,440 @@
+        int gridWidthInPixels = grid.getWidth() * CELL_SIZE;
+        int gridHeightInPixels = grid.getHeight() * CELL_SIZE;
+
+        float offsetX = (Gdx.graphics.getWidth() - gridWidthInPixels) / 2f;
+        float offsetY = (Gdx.graphics.getHeight() - gridHeightInPixels) / 2f;
+
+        spriteBatch.draw(appleTexture,
+                offsetX + apple.getX() * CELL_SIZE,
+                offsetY + apple.getY() * CELL_SIZE,
+                CELL_SIZE,
+                CELL_SIZE);
+    }
+
+    /**
+     * Renders the snake on the grid using the snake texture.
+     */
+    private void renderSnake() {
+        int gridWidthInPixels = grid.getWidth() * CELL_SIZE;
+        int gridHeightInPixels = grid.getHeight() * CELL_SIZE;
+
+        float offsetX = (Gdx.graphics.getWidth() - gridWidthInPixels) / 2f;
+        float offsetY = (Gdx.graphics.getHeight() - gridHeightInPixels) / 2f;
+
+        // Get the current direction of the snake
+        Direction direction = snake.getDirection();
+        float rotation = 0f;
+
+        switch (direction) {
+            case UP -> rotation = 180f;
+            case DOWN -> rotation = 0f;
+            case LEFT -> rotation = 270f;
+            case RIGHT -> rotation = 90f;
+            default -> {
+            }
+        }
+
+        // Render snake head with rotation
+        spriteBatch.draw(
+                snakeTexture,
+                offsetX + snake.getX() * CELL_SIZE,  // x position
+                offsetY + snake.getY() * CELL_SIZE,  // y position
+                CELL_SIZE / 2f,                      // originX (center of the texture)
+                CELL_SIZE / 2f,                      // originY (center of the texture)
+                CELL_SIZE,                           // width
+                CELL_SIZE,                           // height
+                1f,                                  // scaleX
+                1f,                                  // scaleY
+                rotation,                            // rotation in degrees
+                0,                                   // srcX (region's x-coordinate)
+                0,                                   // srcY (region's y-coordinate)
+                snakeTexture.getWidth(),             // srcWidth (width of the texture)
+                snakeTexture.getHeight(),            // srcHeight (height of the texture)
+                false,                               // flipX
+                false                                // flipY
+        );
+
+        // Render snake body
+        for (Snake.Segment segment : snake.getBodySegments()) {
+            spriteBatch.draw(snakeBody, offsetX + segment.getX() * CELL_SIZE,
+                    offsetY + segment.getY() * CELL_SIZE, CELL_SIZE, CELL_SIZE);
+        }
+    }
+
+    private void renderGameScore() {
+        spriteBatch.begin();
+
+        // Score location (center)
+//        int offsetScoreX = (Gdx.graphics.getWidth() / 2) - 130;
```

Comment:
remove commented code

Reply:
Kept for potential design changes with future integration of buttons on screen

Reply:
See above